noir-projects/noir-contracts/contracts/token_contract/src/main.nr

```diff
@@ -64,7 +62,7 @@ contract Token {
         minters: Map<AztecAddress, PublicMutable<bool>>,
         // docs:end:storage_minters
         // docs:start:storage_balances
-        balances: Map<AztecAddress, BalanceSet<TokenNote>>,
+        balances: Map<AztecAddress, BalanceSet<IntNote>>,
```

Comment:
Why make `BalanceSet` generic? If `UintNote` was embedded inside of it (as it maybe should, since it's a balance of uints) the entire indirect dispatch via the `OwnedNote` trait would go away.

Reply:
I guess in principle `BalanceSet` could be reused with a slightly different kind of note that held the amount plus some other stuff, but it's not like we're aiming for a reusable library of these components at this stage. We don't even really have a way to import this.